x-pack/legacy/plugins/graph/public/components/graph_save_modal.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { EuiFormRow, EuiTextArea, EuiCallOut, EuiSpacer, EuiSwitch } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  SavedObjectSaveModal,
+  OnSaveProps,
+} from '../../../../../../src/plugins/kibana_react/public';
+
+import { GraphSavePolicy } from '../types/config';
+
+export interface OnSaveGraphProps extends OnSaveProps {
+  newDescription: string;
+  dataConsent: boolean;
+}
+
+export function GraphSaveModal({
+  onSave,
+  onClose,
+  title,
+  description,
+  showCopyOnSave,
+  savePolicy,
+  hasData,
+}: {
+  onSave: (props: OnSaveGraphProps) => void;
+  onClose: () => void;
+  title: string;
+  description: string;
+  showCopyOnSave: boolean;
+  savePolicy: GraphSavePolicy;
+  hasData: boolean;
+}) {
+  const [newDescription, setDescription] = useState(description);
+  const [dataConsent, setDataConsent] = useState(false);
+  return (
+    <SavedObjectSaveModal
+      onSave={props => {
+        onSave({ ...props, newDescription, dataConsent });
+      }}
+      onClose={onClose}
+      title={title}
+      showCopyOnSave={showCopyOnSave}
+      objectType={i18n.translate('xpack.graph.topNavMenu.save.objectType', {
+        defaultMessage: 'workspace',
+      })}
+      options={
+        <>
+          <EuiFormRow
+            fullWidth
+            label={i18n.translate('xpack.graph.topNavMenu.save.descriptionInputLabel', {
+              defaultMessage: 'Description',
+            })}
+          >
+            <EuiTextArea
+              data-test-subj="dashboardDescription"
+              value={newDescription}
+              placeholder={i18n.translate(
+                'xpack.graph.topNavMenu.save.descriptionInputPlaceholder',
+                { defaultMessage: 'Any notes..' }
```

Comment:
//nit 

This message might be unnecessary.

Reply:
Copied over from the old implementation, but I think you are right, I will remove it